packages/dataviews/src/view-actions.js

```diff
@@ -35,84 +34,81 @@ function ViewTypeMenu( { view, onChangeView, supportedLayouts } ) {
 	}
 	const activeView = _availableViews.find( ( v ) => view.type === v.type );
 	return (
-		<DropdownSubMenu
+		<DropdownMenu
 			trigger={
-				<DropdownSubMenuTrigger
+				<DropdownMenuItem
 					suffix={
-						<>
-							{ activeView.label }
-							<Icon icon={ chevronRightSmall } />
-						</>
+						<span aria-hidden="true">{ activeView.label }</span>
 					}
 				>
-					{ __( 'Layout' ) }
-				</DropdownSubMenuTrigger>
+					<DropdownMenuItemLabel>
+						{ __( 'Layout' ) }
+					</DropdownMenuItemLabel>
+				</DropdownMenuItem>
 			}
+			placement="left-start"
 		>
 			{ _availableViews.map( ( availableView ) => {
 				return (
-					<DropdownMenuItem
+					<DropdownMenuRadioItem
 						key={ availableView.type }
-						role="menuitemradio"
-						aria-checked={ availableView.id === view.type }
-						prefix={
-							availableView.type === view.type && (
-								<Icon icon={ check } />
-							)
-						}
-						onSelect={ () => {
+						value={ availableView.type }
+						name="view-actions-available-view"
+						checked={ availableView.type === view.type }
```

Comment:
Previously, the code was checking `availableView.id === view.type`, but it seems like we should check `availableView.type` instead of `availableView.id`